### PR TITLE
Visemes Creation 

### PR DESCRIPTION
**Pull request by Yusarina**
**Do not merge this unto the translations are merge again my main branch is kinda messed up due to not have translation in it's own branch.** 

This is not ready for prime time, I still need to add stuff and make this system better however we need to get the armture slelection done before I continue. However the viseme creation itself seems to work mostly fine, does need a little bit of work though. This is very much inspired via the Cats one and some of the code has been used form cats, though some of it has been changed.

To-Do
- Mesh auto detection from selected Armature.
- Auto detection of shapekeys on said mesh.
- Translations.
- Maybe some optimizations on the creation.

### CODE REVIEW

**Comment by Yusarina** [comment]
Fixed Conflicts

**Comment by 989onan** [comment]
This needs a revisit, we have translations now so this may need conflicts resolved and integrated with translations

File: core/common.py [hunk 1]
```diff
@@ -114,4 +114,76 @@ def duplicatebone(b: bpy.types.EditBone) -> bpy.types.EditBone:
     cb.tail = b.tail
     cb.matrix = b.matrix
     cb.parent = b.parent
-    return cb
\ No newline at end of file
+    return cb
+
+def has_shapekeys(mesh_obj: Object) -> bool:
```
Comment (989onan) [inline core/common.py:119]: this is instanciated twice, must have done that in the chaos

Comment (989onan) [reply core/common.py:119]: other than this, looks good!

**Review by 989onan — CHANGES_REQUESTED** [review]
Reviewed, needs a small change

**Comment by Yusarina** [comment]
Oh lol, I already was making changes lol 

- Viesmes will now use selected armature.
- New dropdown menu in the viseme UI so the user can select which mesh to create visemes on.
- New helper function get_armature_meshes
- Added a new check before we create the new visemes to see if any exist, if there do we will remove them and create the new ones.
- fixed several issues and errors.

**Review by 989onan — APPROVED** [review]
looks good, perfect!

**Pull request merged**